packages/x-tree-view/src/hooks/useTreeItem2Utils/useTreeItem2Utils.tsx

```diff
@@ -137,16 +143,27 @@ export const useTreeItem2Utils = ({
     event: React.SyntheticEvent & MuiCancellableEvent,
     label: string,
   ) => {
-    if (instance.updateItemLabel && label) {
+    // As a side effect of `instance.focusItem` called here and in `handleCancelItemLabelEditing` the `labelInput` is blurred
+    // The `onBlur` event is triggered, which calls `handleSaveItemLabel` again.
+    // To avoid creating an unwanted behavior we need to check if the item is being edited before calling `updateItemLabel`
+    // using `instance.isItemBeingEditedRef` instead of `instance.isItemBeingEdited` since the state is not yet updated in this point
+    if (
+      instance.updateItemLabel &&
```

Comment:
We do you need to check for `instance.updateItemRef` but not `instance/isItemBeingEditedRef` existence?
I think storing `isEditingEnabled` in a variable defeats the type narrowing done by `hasUtil`